ntoskrnl/io/pnpmgr/devaction.c

```diff
@@ -1002,16 +969,19 @@ IopQueryHardwareIds(PDEVICE_NODE DeviceNode,
                                &ValueName,
                                0,
                                REG_MULTI_SZ,
-                               (PVOID)IoStatusBlock.Information,
+                               (PVOID)HwId,
                                (TotalLength + 1) * sizeof(WCHAR));
         if (!NT_SUCCESS(Status))
         {
             DPRINT1("ZwSetValueKey() failed (Status %lx)\n", Status);
         }
+
+        if (HwId)
+            ExFreePool(HwId);
```

Comment:
As I currently understand the `PiIrpQueryPnPDeviceId` routine initializes the `Id` pointer into the stack, I do not see anywhere that the routine allocates this pointer in a pool. Are you sure you really intend to free the pool there?

Reply:
`HwId` allocated by some function called by `PiIrpQueryPnPDeviceId` (`IRP_MN_QUERY_ID` handling by some driver called by `IopSynchronousCall`).

Reply:
A driver returns ID(s) in response to the IRP, it allocates a WCHAR structure from paged pool to contain the ID(s). The PnP manager frees the structure when it is no longer needed.